### ISSUE TITLE
Allow bypassing UMAP for low-dim (not just 2D) deep embeddings

### ISSUE THREAD

**Issue by PipaFlores** [PipaFlores/encoder-pacman#37]
## Problem
`PatternAnalysis.fit()` only skips the `reducer` (UMAP by default) step when the embedder's output is 2-dimensional:

```python
if self.embeddings.shape[1] > 2:
    self.reduced_embeddings = self.reducer.fit_transform(self.embeddings)
else:
    self.reduced_embeddings = self.embeddings
```
(see [pattern_analysis.py](src/analysis/pattern_analysis.py))

This means an embedder configured with `latent_dimension=5` (e.g. to directly produce a small clusterable space, as discussed for the new MLP baseline in #35) still gets passed through UMAP, even though the intent of choosing a small latent dim directly was often to avoid that extra reduction pass.

## Proposal
Make the "skip the reducer" decision explicit rather than hardcoded to `<= 2`, e.g.:
- A `skip_reducer` / `reducer=None` option that bypasses `self.reducer.fit_transform(...)` regardless of `latent_dimension`, or
- A configurable dimensionality threshold instead of the hardcoded `2`.

## Scope
- Only affects the reduction step in `fit()`; embedders themselves are unaffected (any `latent_dimension` already works).
- Should keep the current default behavior (2D embeddings bypass the reducer) unchanged unless explicitly configured otherwise.
